src/RoslynPad.Common.UI/ViewModels/DocumentViewModel.cs

```diff
@@ -160,17 +185,80 @@ private ObservableCollection<DocumentViewModel> ReadChildren()
         {
             return new ObservableCollection<DocumentViewModel>(
                 IOUtilities.EnumerateDirectories(Path)
-                .Select(x => new DocumentViewModel(x, isFolder: true))
-                .OrderBy(OrderByName)
-                    .Concat(IOUtilities.EnumerateFiles(Path, "*" + DefaultFileExtension)
-                        .Select(x => new DocumentViewModel(x, isFolder: false))
-                        .Where(x => !x.IsAutoSave)
-                        .OrderBy(OrderByName)));
+                    .Concat(IOUtilities.EnumerateFiles(Path, "*" + DefaultFileExtension))
+                    .Select(x => new DocumentViewModel(x, _documentFileWatcher))
+                    .Where(x => !x.IsAutoSave)
+                    .OrderBy(dd => !dd.IsFolder)
+                    .ThenBy(OrderByName));
         }
```

Comment:
I prefer the previous way, where directories and files were sorted separately, and it allows for directory and file objects to be built differently. Plus it saves some IO by not checking `IsDirectory` in the ctor when it's already known.